### PR TITLE
Add transfer with sig functions

### PR DESCRIPTION
**Pull request by nglglhtr**
- #68 
- new `matic.getTransferSignature(
    toSell, toBuy, {
      from: account
    })` function call
  - returns a signed eip712 typed signature for [`TokenTransferOrder` ](https://github.com/maticnetwork/contracts/blob/6413308db75ecdbf8ab9ec2beee1db0d362acea3/contracts/child/misc/LibTokenTransferOrder.sol#L11)
- new 
    ```
    matic.transferWithSignature(
        token,   // token that will be transferred
        sig,     // signature with the intent to buy tokens
        toSell,  // sell order struct
        toBuy,  // buy order struct
        orderFiller, // order fulfiller
        {
          from: acc3 // approved spender
        }
      )
    ```
    function
  - takes a valid eip712 typed `sig` and transfers the `token` to `orderFiller`
  - executes [transferWithSig](https://github.com/maticnetwork/contracts/blob/6fde010694508cd56e43d57c75acd17d2b1878ef/contracts/child/BaseERC20.sol#L37) on child token

### CODE REVIEW

**Comment by nglglhtr** [comment]
a few things: 
- Is there a good way to write https://github.com/maticnetwork/matic.js/blob/be02e63cf94184b63a5ad09f386fce154209f482/src/common/ContractsBase.ts#L28 for now i've added transferWithSig specific ABI in `/src/common/`, I wanted a single function to work for both tokens for simplicity
- do we need a separate call to `Marketplace` contract? [It is just two calls to transferWithSig](https://github.com/maticnetwork/contracts/blob/6fde010694508cd56e43d57c75acd17d2b1878ef/contracts/child/misc/Marketplace.sol#L31), essentially swapping the tokens; would a sample implementation in an example file be better in this case?

File: src/common/ContractsBase.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import BN from 'bn.js'
 import ChildERC20Artifact from 'matic-protocol/contracts-core/artifacts/ChildERC20.json'
 import ChildERC721Artifact from 'matic-protocol/contracts-core/artifacts/ChildERC721.json'
 import ChildMaticArtifact from 'matic-protocol/contracts-core/artifacts/MaticChildERC20.json'
-
+import transferWithSigABI from './transferWithSigABI.json'
```
Comment (ptsayli) [inline src/common/ContractsBase.ts:6]: No need to add this, You can access `transferWithSig` from `ChildERC721Artifact `

File: src/common/Utils.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+export class Utils {
+  getOrderHash(order) {
+    const orderData = Buffer.concat([
+      EthUtil.toBuffer(order.id),
+      EthUtil.toBuffer(order.token),
+      EthUtil.setLengthLeft(order.amount, 32),
+    ])
+    return EthUtil.keccak256(orderData)
+  }
+  getTypedData({ token, spender, tokenIdOrAmount, data, expiration, chainId }) {
```
Comment (ptsayli) [inline src/common/Utils.ts:13]:
```suggestion

    getTypedData({ token, spender, tokenIdOrAmount, data, expiration, chainId }) {
```

File: examples/execute-order.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+const bn = require('bn.js')
+const utils = require('./utils')
+
+const privateKey1 = '0x..'
```
Comment (ptsayli) [inline examples/execute-order.js:4]: Add `privatekey` safety warning comment here

File: examples/execute-order.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+const privateKey2 = '0x..'
+
+async function init() {
+  const { matic, wallet } = await utils.getMaticClient('testnet', 'v3')
+
+  const token1 = '0x..' // address of token to sell
```
Comment (ptsayli) [inline examples/execute-order.js:10]: would be better to add sample/test token address, amount etc.
User should able to run example code with minimum changes

File: src/common/ContractsBase.ts (hunk 1 above)
Comment (nglglhtr) [reply src/common/ContractsBase.ts]: worked, thanks!

**Pull request merged**